test/simulator-e2e-specs.js

```diff
@@ -360,6 +361,74 @@ function runTests (deviceType) {
       await getTouchEnrollKeys().should.eventually.deep.equal(originalValues);
     });
   });
+
+  describe(`multiple instances of ${deviceType.version} simulator on Xcode9+`, function () {
+    this.timeout(LONG_TIMEOUT);
+    const simulatorsMapping = new Map();
+    const DEVICES_COUNT = 2;
+    const verifyStates = async (sim, shouldServerRun, shouldClientRun) => {
+      const isServerRunning = await sim.isRunning();
+      isServerRunning.should.eql(shouldServerRun);
+      const isClientRunning = await sim.isUIClientRunning();
+      isClientRunning.should.eql(shouldClientRun);
+    };
+
+    before(async function () {
+      const xcodeVersion = await xcode.getVersion(true);
+      if (xcodeVersion.major < 9) {
+        return this.skip();
+      }
+
+      await killAllSimulators();
+      for (const deviceIdx of _.range(1, DEVICES_COUNT + 1)) {
+        const udid = await simctl.createDevice(`ios-simulator_${deviceIdx}_testing`,
+                                               deviceType.device,
+                                               deviceType.version);
+        const sim = await getSimulator(udid);
+        simulatorsMapping.set(udid, sim);
+      }
+    });
+    after(async function () {
+      const xcodeVersion = await xcode.getVersion(true);
+      if (xcodeVersion.major < 9) {
+        return this.skip();
+      }
+
+      const existingDevices = (await simctl.getDevices())[deviceType.version];
```

Comment:
it's to make sure all the devices, we tried to create in _before_ method, exist and ready for removal. There is a possibility they don't exist if _before_ method failed.